src/LibYear.Lib/FileTypes/XmlProject.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Xml.Linq;
+using NuGet.Versioning;
+
+namespace LibYear.Lib.FileTypes
+{
+    public abstract class XmlProject : IProjectFile
+    {
+        public virtual string FileName => throw new NotSupportedException();

```

Comment:
It's virtual because whilst it doesn't support FileName, inherited classes do implement it (and is required for the interface).